### ISSUE TITLE
Best way to version an rc release (`rc1`, `.rc1`, `-rc1`)?

### ISSUE THREAD

**Issue by yarikoptic** [dandi/dandi-cli#179]
- semver which we say we use [wants](https://semver.org/spec/v2.0.0.html) `-rc.1`
- [pip documentation](https://pip.pypa.io/en/stable/user_guide/) says to use semantic versioning in general and no word about pre-release candidates
- in datalad in recent ones we used `rc1` (so I know that it generally "works") which seems to be inline with good/old/odd Python's approach as also supported by `packaging`: https://packaging.pypa.io/en/latest/version/
- although https://packaging.python.org/guides/distributing-packages-using-setuptools/#pre-release-versioning talks about `.rc1`...

uff.  What scheme would you @jwodder recommend so we play most nicely with all the kids in the neighborhood? ;) (I guess the best approach is to avoid any `rc`)

**Comment by jwodder** [comment]
[PEP 440](https://www.python.org/dev/peps/pep-0440/) (the specification for the Python project version format, which `packaging` implements) specifies that the standard spelling for release candidates is "`<base version>rc<integer>`".  The insertion of a period, hyphen, or underscore before the `rc` is allowed as an alternative syntax which normalizes to the no-punctuation form.  I would recommend the no-punctuation form.

**Comment by yarikoptic** [comment]
ok, let's proceed with `0.6.0rc1` and see if some semver gods would not rain the fire upon us.  Thanks!

**Comment by satra** [comment]
@yarikoptic - there is a difference between PEP440 and semver.org, but i think across most python and other projects we have been reasonable about keeping a happy medium. just to provide consistency with most other ni* projects, perhaps just stick to the semver version 0.6.0-rc1. i know the hyphen isn't necessary in PEP440, but keeps this compatible with both.

**Comment by yarikoptic** [comment]
Apparently even `rc1` displeases  distutils anyways, so if pip/pypi would digest well `-rc1`, why not... will do
```
$> python -c 'from distutils.version import LooseVersion as LV, StrictVersion as SV; print(SV("0.6.0rc1") < "0.6.0")'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python2.7/distutils/version.py", line 40, in __init__
    self.parse(vstring)
  File "/usr/lib/python2.7/distutils/version.py", line 107, in parse
    raise ValueError, "invalid version number '%s'" % vstring
ValueError: invalid version number '0.6.0rc1'
```

**Comment by yarikoptic** [comment]
FWIW, 
- tagged and pushed 0.6.0-rc1 in `0.6.x` branch.
- `make release-pypi` uploaded as 0.6.0rc1 version to pypi
- it seems to install and operate ok whenever exact version is specified.

- I thought that I could use `pip3 install --pre dandi~=0.6.0` to get it installed without specifying specific release candidate but pinning to this upcoming release -- that doesn't work.
  - note that IIRC installing a pre-release opens the "door" for pre-releases to be allowed to be installed for future releases (e.g. `0.7.0`) .  That is why I was trying to pin
- `pip install --pre dandi` works or less preferable since we might get another `rc`: `pip install dandi==0.6.0rc1`

**Comment by satra** [comment]
if we do release `0.6.0-rc1` we can remove `0.6.0rc1` from pypi. also we should just setup releases also from github actions if we haven't already (many examples out there :) ).

**Comment by yarikoptic** [comment]
We have "released" it as far as it goes, now will do on github marking it as "pre-production": done now https://github.com/dandi/dandi-cli/releases .

The fact that it uploaded it to pypi without `-` I took as a "feature" of some kind (e.g. sanitization to match pypi expectations etc).  Since it is a "rc" I would not sweat much to harmonize it further unless you see some major side-effect we would like to avoid.

**Comment by yarikoptic** [comment]
re automation -- probably should not hurt.  I did something like that for `duecredit` and I also enjoyed some cons, e.g. if tag was not pushed along with the branch progress, then IIRC it did not build/upload... but may be it was a shortcoming of my recipe and it could be done more "natively" now using github workflows to react explicitly on releases.  But then it is automated but would incur delay, thus complicating testing if compared to current "manual make" call.
